slides/hash_functions.tex

```diff
@@ -0,0 +1,242 @@
+% Choose one to switch betweeen slides and handout
+%\documentclass[]{beamer}
+\documentclass[handout]{beamer}
+
+% Video Meta Data
+\title{Bitcoin, Blockchain and Cryptoassets}
+\subtitle{Introduction to Cryptography: Hash Functions}
+\author{Prof. Dr. Fabian Schär}
+\institute{University of Basel}
+
+% Config File
+\input{../config/config.tex}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{document}
+
+\thispagestyle{empty}
+\begin{frame}[noframenumbering]
+	\titlepage
+\end{frame}
+
+%%%
+\begin{frame}{What Is a Hash Function?}
+
+Deterministic algorithm (\color{focus}function, $H()$\color{black}) that maps data of quasi-arbitrary size (\color{focus}pre-image, $m$\color{black}) to fixed-length bit string (\color{focus}hash value, $h$\color{black}).
+
+	\begin{align}
+		h = H(m)
+		\label{eq:hash_function}
+	\end{align}
+	\vspace{1em}
+	
+\uncover<1->{Example of a simplified hash function:
```

Comment:
Before we go into the example, we should quickly introduce mod operations. Let's use the mod wheel from the book / original lecture and show some simple examples.

Reply:
Proposal for implementation included in commit #99343a9